assembly/assembly-wsmaster-war/src/main/java/org/eclipse/che/api/deploy/WsMasterModule.java

```diff
@@ -150,6 +151,7 @@ protected void configure() {
     Multibinder<FactoryParametersResolver> factoryParametersResolverMultibinder =
         Multibinder.newSetBinder(binder(), FactoryParametersResolver.class);
     factoryParametersResolverMultibinder.addBinding().to(GithubFactoryParametersResolver.class);
+    factoryParametersResolverMultibinder.addBinding().to(BitbucketServerFactoryParametersResolver.class);
```

Comment:
what about not add bitbucket at all, when it is not set in configuration?

Reply:
Conditional binding in general is an antipattern of guice. Moreover, it's hard/impossible to do in general(only if you rely on env variables.).